internal/gamepaddb/gen.go

```diff
@@ -0,0 +1,183 @@
+
+{{.BuildTag}}
+
+package gamepaddb
+
+import _ "embed"
+
+//go:embed gamecontrollerdb_{{.Platform}}.txt
+var controllerBytes []byte
+{{if eq .Platform "windows" }} 
+//go:embed gamecontrollerdb_glfw.txt
+var glfwControllerBytes []byte
+{{end}}
+func init() {
+	if err := Update(controllerBytes); err != nil {
+		panic(err)
+	}
+{{if eq .Platform "windows" }} 
+    if err := Update(glfwControllerBytes); err != nil {
+        panic(err)
+    }
+{{end}}}`
+
+func main() {
+	// Follow the standard comment rule (https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source).
+	doNotEdit := "// Code generated by gen.go using 'go generate'. DO NOT EDIT."
+
+	type gamePadPlatform struct {
+		name     string
+		buildTag string
+	}
+
+	supported := map[string]gamePadPlatform{
+		"Windows": {
+			name:     "windows",
+			buildTag: "//go:build windows && !microsoftgdk",
+		},
+		"Mac OS X": {
+			name:     "macos_darwin",
+			buildTag: "//go:build darwin && !ios",
+		},
+		"Linux": {
+			name:     "linbsd",
+			buildTag: "//go:build (freebsd || (linux && !android) || netbsd || openbsd) && !nintendosdk && !playstation5",
+		},
+		"iOS": {
+			name:     "ios",
+			buildTag: "//go:build ios",
+		},
+		"Android": {
+			name:     "android",
+			buildTag: "//go:build android",
+		},
+	}
+
+	// divide controller dbs into chunks by platform
+	controllerDbs := splitControllerDb(gamecontrollerdb_txt)
+
+	for sdlName, platform := range supported {
+		controllerDb, ok := controllerDbs[sdlName]
+		if !ok {
+			log.Fatalf("failed to find controller db for platform %s in gamecontrollerdb_txt", sdlName)
+		}
+
+		// write each chunk into separate text file for embedding into respective generated files
+		txtFile, err := os.Create(fmt.Sprintf("gamecontrollerdb_%s.txt", platform.name))
+		if err != nil {
+			log.Fatalf("failed to open file: %v", err)
+		}
+		defer txtFile.Close()
+		written, err := txtFile.Write(controllerDb)
+		if err != nil {
+			log.Fatalf("failed to write controller db for %s, expected to write %d bytes, wrote %d: %v", sdlName, len(controllerDb), written, err)
+		}
+
+		path := fmt.Sprintf("db_%s.go", platform.name)
+		tmpl, err := template.New(path).Parse(db_template)
+		if err != nil {
+			log.Fatalf("failed to parse template: %v", err)
+		}
+
+		f, err := os.Create(path)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer f.Close()
+
+		err = tmpl.Execute(f, struct {
+			License   string
+			DoNotEdit string
+			BuildTag  string
+			Platform  string
+		}{
+			License:   license,
+			DoNotEdit: doNotEdit,
+			BuildTag:  platform.buildTag,
+			Platform:  platform.name,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+// splitControllerDb maps the game controller db into respective controller definition byte slices for
+// each platform
+func splitControllerDb(controllerDb []byte) map[string][]byte {
+	dbs := map[string][]byte{}
+	// split at #, skip first 3 header chunks
+	chunks := bytes.Split(controllerDb, []byte{'#'})[3:]
```

Comment:
Just skipping certain amount of lines seems fragile.

Let's use `bytes.Reader` and `bufio.Scanner` to process each line.